opennlp-tools/src/main/java/opennlp/tools/ml/model/OnePassDataIndexer.java

```diff
@@ -39,141 +39,175 @@
+		LinkedList<Event> events = new LinkedList<>();
+		Event ev;
+		while ((ev = eventStream.read()) != null) {
+			events.addLast(ev);
+			update(ev.getContext(), predicateSet, counter, cutoff);
+		}
+		predCounts = new int[predicateSet.size()];
+		int index = 0;
+		for (Iterator<String> pi = predicateSet.iterator(); pi.hasNext(); index++) {
+			String predicate = pi.next();
+			predCounts[index] = counter.get(predicate);
+			predicatesInOut.put(predicate, index);
+		}
+		return events;
+	}
+
+	protected List<ComparableEvent> index(LinkedList<Event> events,
+			Map<String, Integer> predicateIndex) {
+		Map<String, Integer> omap = new HashMap<>();
+
+		int numEvents = events.size();
+		int outcomeCount = 0;
+		List<ComparableEvent> eventsToCompare = new ArrayList<>(numEvents);
+		List<Integer> indexedContext = new ArrayList<>();
+
+		for (int eventIndex = 0; eventIndex < numEvents; eventIndex++) {
+			Event ev = events.removeFirst();
+			String[] econtext = ev.getContext();
+			ComparableEvent ce;
+
+			int ocID;
+			String oc = ev.getOutcome();
+
+			if (omap.containsKey(oc)) {
+				ocID = omap.get(oc);
+			} else {
+				ocID = outcomeCount++;
+				omap.put(oc, ocID);
+			}
+
+			for (String pred : econtext) {
+				if (predicateIndex.containsKey(pred)) {
+					indexedContext.add(predicateIndex.get(pred));
+				}
+			}
+
+			// drop events with no active features
+			if (indexedContext.size() > 0) {
+				int[] cons = new int[indexedContext.size()];
+				for (int ci = 0; ci < cons.length; ci++) {
+					cons[ci] = indexedContext.get(ci);
+				}
+				ce = new ComparableEvent(ocID, cons);
+				eventsToCompare.add(ce);
+			} else {
+				System.err.println("Dropped event " + ev.getOutcome() + ":"
+						+ Arrays.asList(ev.getContext()));
+			}
+			// recycle the TIntArrayList
+			indexedContext.clear();
+		}
+		outcomeLabels = toIndexedStringArray(omap);
+		predLabels = toIndexedStringArray(predicateIndex);
+		return eventsToCompare;
+	}
+
+	/**
+	 * One argument constructor for DataIndexer which calls the two argument
+	 * constructor assuming no cutoff.
+	 *
+	 * @param eventStream
+	 *          An Event[] which contains the a list of all the Events seen in the
+	 *          training data.
+	 */
+	@Deprecated
+	public OnePassDataIndexer(ObjectStream<Event> eventStream) throws IOException {
+		this(eventStream, 0);
+	}
+
+	@Deprecated
+	public OnePassDataIndexer(ObjectStream<Event> eventStream, int cutoff)
+			throws IOException {
+		this(eventStream, cutoff, true);
+	}
+
+	/**
+	 * Two argument constructor for DataIndexer.
+	 *
+	 * @param eventStream
+	 *          An Event[] which contains the a list of all the Events seen in the
+	 *          training data.
+	 * @param cutoff
+	 *          The minimum number of times a predicate must have been observed in
+	 *          order to be included in the model.
+	 */
+	@Deprecated
+	public OnePassDataIndexer(ObjectStream<Event> eventStream, int cutoff, boolean sort)
```

Comment:
- Moving code makes it harder to review, it could have stayed where it was, then it would just be a matter of one line insert for the deprecation.